rust/lance/src/dataset/write/merge_insert.rs

```diff
@@ -428,7 +508,7 @@ impl MergeInsertJob {
             HashJoinExec::try_new(
                 shared_input,
                 target,
-                vec![(Arc::new(target_key), Arc::new(source_key))],
+                vec![(Arc::new(source_key), Arc::new(target_key))],
```

Comment:
Ah, good catch.  I guess we got away with it in the past since the position of the key field was always equal in the two schemas?

Reply:
It seems so.